docs/content/en/docs/pipeline-stages/builders/docker.md

```diff
@@ -67,15 +67,37 @@ These credentials are configured in the `cluster` section with the following opt
 
 {{< schema root="ClusterDetails" >}}
 
-To set up the credentials for Kaniko refer to the [kaniko docs](https://github.com/GoogleContainerTools/kaniko#kubernetes-secret) (**Note**: Rename the downloaded JSON key to *kaniko-secret* without appending *.json*).
-Alternatively, the path to a credentials file can be set with the `pullSecret` option:
+### Configure Kaniko Credentials 
+
+To set up the credentials for Kaniko refer to the [kaniko docs](https://github.com/GoogleContainerTools/kaniko#kubernetes-secret)
+
+(**Note**: Rename the downloaded JSON key to *kaniko-secret* without appending *.json*).
```

Comment:
Should this be placed in the artifact's `context` directory?

Reply:
no. i added a note down that please do not check in secret in the directory. This secret could be at any location.

Reply:
Sorry, I'm not very adept remote cluster builds with Kaniko, and I'm confused by both this section and the linked to Kaniko secrets.  The linked-to Kaniko document uses kubectl to create a secret wrapping the service-account key file to access GCR.

Would it make sense to just copy the relevant steps here?

It's also not clear to me where I'd put this `kaniko-secret` file.  It seems there's a standard place, given the "Alternatively, …" sentence below.

Reply:
No wonder it's confusing.

When creating a secret with
```
kubectl create secret generic kaniko-secret --from-file=<path to kaniko-secret.json>
```
the key name of the `data:` map in the created secret is set to the filename given by `--from-file`. So, if you follow the linked Kaniko doc, you'll end up having the following for the secret.
```yaml
data:
  kaniko-secret.json: <... secret data ...>
``` 
However, Skaffold by default [expects](https://github.com/GoogleContainerTools/skaffold/issues/3828#issuecomment-626965077) you to create the secret with the key name `kaniko-secret`. That's the reason for the instruction to rename the JSON key file to `kaniko-secret` without `.json`.

Reply:
> It's also not clear to me where I'd put this `kaniko-secret` file. It seems there's a standard place, given the "Alternatively, …" sentence below.

We could copy the instructions here instead of pointing to kaniko docs.

Reply:
I agree that we should copy the instructions to these docs. It makes for a much easier read through them